oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticTestServer.java

```diff
@@ -44,23 +44,32 @@ public class ElasticTestServer implements AutoCloseable {
 
     private static final Logger LOG = LoggerFactory.getLogger(ElasticTestServer.class);
     private static final String PLUGIN_DIGEST = "db479aeee452b2a0f6e3c619ecdf27ca5853e54e7bc787e5c56a49899c249240";
-    private static ElasticTestServer esTestServer;
-    private volatile ElasticsearchContainer elasticsearchContainer;
+    private static ElasticTestServer esTestServer = new ElasticTestServer();
+    private static volatile ElasticsearchContainer elasticsearchContainer;
 
-    public static synchronized ElasticTestServer getESTestServer() {
-            if (esTestServer == null) {
-                LOG.info("Starting ES test server");
-                esTestServer = new ElasticTestServer();
-                esTestServer.setup();
-                Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-                    LOG.info("Stopping global ES test server.");
-                    esTestServer.close();
-                }));
+    private ElasticTestServer() {
+    }
+
+    public static synchronized ElasticsearchContainer getESTestServer() {
+        // Setup a new ES container if elasticsearchContainer is null or not running
+        if (elasticsearchContainer == null || !elasticsearchContainer.isRunning()) {
```

Comment:
Can a container be in different state like starting?

Reply:
yes , there's a created state. But elasticsearchContainer.start(); in the setup already retries untill the container gets to running state. 
So if till here, we don't have the container in running state, there's no use to wait more.